README.md

```diff
@@ -195,123 +197,284 @@ Let's break down the example above:
-    ///   - arguments: The arguments to pass to the executable.
-    ///   - environment: The environment in which to run the executable.
-    ///   - workingDirectory: The working directory in which to run the executable.
-    ///   - platformOptions: The platform specific options to use
-    ///     when running the executable.
-    ///   - body: The custom execution body to manually control the running process
-    /// - Returns a ExecutableResult type containing the return value
-    ///     of the closure.
-    public static func run<Result>(
-        _ executable: Executable,
-        arguments: Arguments = [],
-        environment: Environment = .inherit,
-        workingDirectory: FilePath? = nil,
-        platformOptions: PlatformOptions = PlatformOptions(),
-        isolation: isolated (any Actor)? = #isolation,
-        _ body: (@escaping (
-                    Subprocess.Execution<CustomWriteInput, RedirectedOutput, RedirectedOutput>,
-                    StandardInputWriter
-                ) async throws -> Result
-            )
-    ) async throws -> ExecutionResult<Result>
-
-    /// Run a executable with given parameters specified by a
-    /// `Subprocess.Configuration`
-    /// - Parameters:
-    ///   - configuration: The `Subprocess` configuration to run.
-    ///   - output: The method to use for redirecting the standard output.
-    ///   - error: The method to use for redirecting the standard error.
-    ///   - body: The custom configuration body to manually control
-    ///       the running process and write to its standard input.
-    /// - Returns a ExecutableResult type containing the return value
-    ///     of the closure.
-    public static func run<Result>(
-        _ configuration: Configuration,
-        isolation: isolated (any Actor)? = #isolation,
-        _ body: (@escaping (
-                    Subprocess.Execution<CustomWriteInput, RedirectedOutput, RedirectedOutput>,
-                    StandardInputWriter
-                ) async throws -> Result
-            )
-    ) async throws -> ExecutionResult<Result>
-}
+/// Run a executable with given parameters and a custom closure
+/// to manage the running subprocess' lifetime and its IOs.
+/// - Parameters:
+///   - executable: The executable to run.
+///   - arguments: The arguments to pass to the executable.
+///   - environment: The environment in which to run the executable.
+///   - workingDirectory: The working directory in which to run the executable.
+///   - platformOptions: The platform specific options to use
+///     when running the executable.
+///   - input: The input to send to the executable.
+///   - output: The method to use for redirecting the standard output.
+///   - error: The method to use for redirecting the standard error.
+///   - body: The custom execution body to manually control the running process
+/// - Returns a CollectedResult containing the result of the run.
+@available(macOS 9999, *)
```

Comment:
What kind of availability do we really need here? I assume it's 9999 because we don't have a macOS with `Span` yet?

Reply:
Remove unnecessary ones and just comment on the proposal that's it's Span.